frontend/packages/core/src/text.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from "react";
+import styled from "@emotion/styled";
+import { Fab, Grid } from "@material-ui/core";
+
+import { ClipboardButton } from "./button";
+
+const CopyButtonContainer = styled(Grid)({
+  marginLeft: "7px",
+});
+
+const Pre = styled.pre({
+  border: "1px solid rgba(13, 16, 48, 0.38)",
+  backgroundColor: "rgba(13,16,48,0.12)",
+  borderRadius: "4px",
+  fontSize: "16px",
+  color: "#242b8c",
+  padding: "12px 16px",
+  flex: 1,
+  whiteSpace: "pre-wrap",
+  wordWrap: "break-word",
+  flexDirection: "row-reverse",
+  display: "flex",
+  overflowY: "scroll",
+});
+
+interface CodeProps {
+  children: string;
+  showCopyButton?: boolean;
+}
+
+const Code = ({ children, showCopyButton = true }: CodeProps) => (
+  <Pre>
+    {showCopyButton && (
+      <CopyButtonContainer container justify="flex-end">
+        <Fab variant="round" size="small">
+          <ClipboardButton text={children} />
+        </Fab>
+      </CopyButtonContainer>
+    )}
+    <Grid container justify="flex-start" alignItems="center">
```

Comment:
somethings up with the width here it's wrapping well before the edge 
![Screenshot from 2021-03-17 14-07-08](https://user-images.githubusercontent.com/4712430/111524165-1bd12680-872a-11eb-867c-c5c0ccb4f2e8.png)

Reply:
probably nowrap would also work better so a single long error doesn't render it unreadable

Reply:
yeah the wrapping is off, I can play around with it but nowrap would introduce an x scroll for long errors which I was trying to avoid. Mainly to allow users to not have to scroll around to read a single error but also because the copy button won't scroll with the text.